bench1/q4 vector Parquet query causes "panic: runtime error: index out of range [253] with length 253"

Repro is with super commit f166ba9 and the mgbench [bench1/q4](https://github.com/brimdata/mgbench-private/blob/main/bench1/q4/q4.spq) query.

Test data is `s3://brim-sampledata/mgbench/bench1-from-duckdb-table.parquet`. This query previously ran ok and I've confirmed via binary search that this started failing at commit f166ba9 which is associated with the changes in #5502.

```
$ super -version
Version: v1.18.0-172-gf166ba93

$ SUPER_VAM=1 super -c '
from "bench1-from-duckdb-table.parquet"
| where
    machine_group=="Servers"
    and cpu_wio > float32(0.99)
    and log_time >= 2016-12-01T00:00:00Z
    and log_time < 2017-01-01T00:00:00Z
| summarize
    spikes := count()
    by machine_name
| sort -r spikes
| head 10'

panic: runtime error: index out of range [253] with length 253

goroutine 72 [running]:
github.com/brimdata/super/vector.Or(0xc0005f1440, 0xc0005f19b0)
	/Users/phil/work/super/vector/bool.go:121 +0x125
github.com/brimdata/super/runtime/vam/expr.(*And).eval(0xc000585440, {0xc0010824a0, 0x2, 0x0?})
	/Users/phil/work/super/runtime/vam/expr/logic.go:83 +0x1c7
github.com/brimdata/super/runtime/vam/expr.(*And).Eval.evalBool.func1({0xc0010824a0, 0x2, 0x2})
	/Users/phil/work/super/runtime/vam/expr/logic.go:165 +0x18d
github.com/brimdata/super/vector.Apply(0x10?, 0x74c92f8?, {0xc0010824a0?, 0xc002e19be8?, 0x605fbc5?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/runtime/vam/expr.evalBool(...)
	/Users/phil/work/super/runtime/vam/expr/logic.go:157
github.com/brimdata/super/runtime/vam/expr.(*And).Eval(0xc000585440, {0x74c92f8, 0xc0004dcfc0})
	/Users/phil/work/super/runtime/vam/expr/logic.go:61 +0x125
github.com/brimdata/super/runtime/vam/expr.(*And).Eval(0xc000585620, {0x74c92f8, 0xc0004dcfc0})
	/Users/phil/work/super/runtime/vam/expr/logic.go:61 +0x3b
github.com/brimdata/super/runtime/vam/expr.(*And).Eval(0xc000585800, {0x74c92f8, 0xc0004dcfc0})
	/Users/phil/work/super/runtime/vam/expr/logic.go:61 +0x3b
github.com/brimdata/super/runtime/vam/op.(*Filter).Pull(0xc000585830, 0x0)
	/Users/phil/work/super/runtime/vam/op/filter.go:25 +0x6b
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull(0xc001aa6e70, 0x0?)
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:58 +0x155
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0xc0008909c0)
	/Users/phil/work/super/runtime/vam/op/combine.go:110 +0x3f
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 14
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35
```

It looks like I'm seeing this same panic with [bench3/q1](https://github.com/brimdata/mgbench-private/blob/main/bench3/q1/q1.spq) and [bench3/q4](https://github.com/brimdata/mgbench-private/blob/main/bench3/q4/q4.spq).

--- a/zio/parquetio/vectorreader.go
+++ b/zio/parquetio/vectorreader.go
@@ -319,14 +319,15 @@ func makeNulls(a arrow.Array) *vector.Bool {
 	if len(bytes) == 0 {
 		return nil
 	}
-	bits := make([]uint64, (len(bytes)+7)/8)
+	n := a.Len()
+	bits := make([]uint64, (n+63)/64)
 	bitsAsBytes := reinterpretSlice[byte](bits)
 	copy(bitsAsBytes, bytes)
 	for i := range bits {
 		// Flip bits
 		bits[i] ^= ^uint64(0)
 	}
-	return vector.NewBool(bits, uint32(a.Len()), nil)
+	return vector.NewBool(bits, uint32(n), nil)
 }
 
 func convertSlice[Out, In uint8 | uint16 | uint32 | uint64 | int8 | int16 | int32 | int64 | float32 | float64](in []In) []Out {

